EIPS/eip-1046.md

```diff
@@ -1,87 +1,164 @@
+     */
+    image?: string;
+    
+    /**
+     * One or more URIs each pointing to a resource with mime type `image/*` that represents this token.
+     * If an image is a bitmap, it SHOULD have a width between 320 and 1080 pixels
+     * Images SHOULD have an aspect ratio between 1.91:1 and 4:5 inclusive.
+     */
+    images?: string[];
+    
+    /**
+     * One or more URIs each pointing to a resource with mime type `image/*` that represent an icon for this token.
+     * If an image is a bitmap, it SHOULD have a width between 320 and 1080 pixels, and MUST have a height equal to its width
+     * Images MUST have an aspect ratio of 1:1, and use a transparent background
+     */
+    icons?: string[];
 }
 ```
 
-The token's name() and symbol() getters should be preferred over the name and/or symbol properties in the tokenURI JSON. 
+### Extension to the ERC-721 Metadata Schema
+
+Contracts that implement ERC-721 and use its token metadata URI SHOULD to use the following extension to the metadata URI:
+
+```typescript
+
+interface ERC721TokenMetadataInterop extends ERC721TokenMetadata {
+    /**
+     * Interoperabiliy, to avoid confusion between ERC-721 and EIP-1046 token URIs.
+     **/
+    interop: {
+        /**
+         * This MUST be true if this is ERC-1046 Token Metadata, otherwise, this MUST be false.
+         **/
+        erc1046: boolean;
+        /**
+         * This MUST be true if this is ERC-721 Token Metadata, otherwise, this MUST be false.
+         **/
+        erc721: boolean;
+    };
+}
+```
+
+### Miscellaneous Recommendations
+
+To save gas, it is RECOMMENDED for compliant contracts not to implement the optional `name()`, `symbol()`, or `decimals()` functions, and instead to only include them in the metadata URI. Additionally, if the decimals is `18`, then it is NOT RECOMMENDED to include the `decimals` field.
 
 ## Rationale
-This proposal will make adding metadata to ERC20 tokens straightforward for developers with minimal-to-no disruption to the overall ecosystem. By using the same parameter name and by consolidating the underlying Token JSON Metadata Standard, developers will confidently understand how to add and interpret token metadata between ERC20 and ERC721 tokens.
+
+This makes adding metadata to ERC-20 tokens more straightforward for developers, with minimal to no disruption to the overall ecosystem. Using the same parameter name makes it easier to reuse code.
+
+Additionally, the recommendations not to use ERC-20's `name`, `symbol`, and `decimals` functions save gas.
 
 ## Backwards Compatibility
-This EIP is fully backwards compatible as its implementation simply extends the functionality of ERC20 tokens and is optional.
 
-## Test Cases
-TO-DO
+This EIP is fully backwards compatible as its implementation simply extends the functionality of ERC-20 tokens and is optional.
+
+## Security Considerations
 
-## Implementation
+### SSRF
```

Comment:
The first time an acronym is used it should be expanded.

Reply:
SSRF is a well-established acronym ("Server Side Request Forgery"), and is therefore exempt from this rule.